src/class/dfu/dfu_rt_device.c

```diff
@@ -106,34 +117,67 @@ bool dfu_rtd_control_xfer_cb(uint8_t rhport, uint8_t stage, tusb_control_request
   // Handle class request only from here
   TU_VERIFY(request->bmRequestType_bit.type == TUSB_REQ_TYPE_CLASS);
 
-  switch ( request->bRequest )
+  TU_LOG2("  DFU Request: %s\r\n", tu_lookup_find(&_dfu_request_table, request->bRequest));
+  switch (request->bRequest)
   {
     case DFU_REQUEST_DETACH:
-      tud_control_status(rhport, request);
-      tud_dfu_runtime_reboot_to_dfu_cb();
+    {
+      if (_dfu_state_ctx.state == APP_IDLE)
+      {
+        _dfu_state_ctx.state = APP_DETACH;
+        if ((_dfu_state_ctx.attrs & DFU_FUNC_ATTR_WILL_DETACH_BITMASK) == 1)
```

Comment:
`== 1` must be dropped, it is bitwise, and will return the WILL_DETACH which is (1<<3) = 8

Reply:
Resolved in 2e2dc7b
No longer applicable.